src/Components/ServiceScene/Breakdowns/LabelBreakdownScene.tsx

```diff
@@ -201,12 +212,190 @@ export class LabelBreakdownScene extends SceneObjectBase<LabelBreakdownSceneStat
+        { value: 'grid', label: 'Grid' },
+        { value: 'rows', label: 'Rows' },
+      ],
+      active: 'grid',
+      layouts: [
+        new SceneFlexLayout({
+          direction: 'column',
+          children: [
+            new SceneFlexItem({
+              minHeight: 300,
+              body,
+            }),
+          ],
+        }),
+        new ByFrameRepeater({
+          body: new SceneCSSGridLayout({
+            templateColumns: GRID_TEMPLATE_COLUMNS,
+            autoRows: '200px',
+            children: [
+              new SceneFlexItem({
+                body: new SceneReactObject({
+                  reactNode: <LoadingPlaceholder text="Loading..." />,
+                }),
+              }),
+            ],
+          }),
+          getLayoutChild: getFilterBreakdownValueScene(
+            getLabelValue,
+            query.expr.includes('count_over_time') ? DrawStyle.Bars : DrawStyle.Line,
+            VAR_LABELS
+          ),
+          sortBy,
+          direction,
+          getFilter,
+        }),
+        new ByFrameRepeater({
+          body: new SceneCSSGridLayout({
+            templateColumns: '1fr',
+            autoRows: '200px',
+            children: [
+              new SceneFlexItem({
+                body: new SceneReactObject({
+                  reactNode: <LoadingPlaceholder text="Loading..." />,
+                }),
+              }),
+            ],
+          }),
+          getLayoutChild: getFilterBreakdownValueScene(
+            getLabelValue,
+            query.expr.includes('count_over_time') ? DrawStyle.Bars : DrawStyle.Line,
+            VAR_LABELS
+          ),
+          sortBy,
+          direction,
+          getFilter,
+        }),
+      ],
+    });
+  }
+
+  public getLabelsVariable(): AdHocFiltersVariable {
+    const variable = sceneGraph.lookupVariable(VAR_LABELS, this)!;
+
+    if (!(variable instanceof AdHocFiltersVariable)) {
+      throw new Error('Filters variable not found');
+    }
+
+    return variable;
+  }
+
+  public getFieldsVariable(): AdHocFiltersVariable {
+    const variable = sceneGraph.lookupVariable(VAR_FIELDS, this)!;
+
+    if (!(variable instanceof AdHocFiltersVariable)) {
+      throw new Error('Filters variable not found');
+    }
+
+    return variable;
+  }
+
+  private getExpr(tagKey: string) {
+    const labelsVariable = this.getLabelsVariable();
+    const fieldsVariable = this.getFieldsVariable();
+
+    let labelExpressionToAdd;
+    let fieldExpressionToAdd = '';
+    // `LEVEL_VARIABLE_VALUE` is a special case where we don't want to add this to the stream selector
+    if (tagKey !== LEVEL_VARIABLE_VALUE) {
+      labelExpressionToAdd = { key: tagKey, operator: '!=', value: '' };
+    } else {
+      fieldExpressionToAdd = `| ${LEVEL_VARIABLE_VALUE} != ""`;
+    }
+    const streamSelectors = [...labelsVariable.state.filters, labelExpressionToAdd]
+      .filter((f) => !!f)
+      .map((f) => `${f!.key}${f!.operator}\`${f!.value}\``)
```

Comment:
If these were undefined, would we rather throw an error at runtime, or execute a query that would likely throw an error?

```suggestion
      .map((f) => `${f?.key}${f?.operator}\`${f?.value}\``)
```

Reply:
Oh, I missed this. I wouldn't mind a linting rule to avoid non-null assertions.

Reply:
Well, I do `.filter((f) => !!f)` before. I don't see how `f` would still be `null | undefined` here. Let me fix it with a type predicate, so we can spare both `!` and `?`.

Reply:
WDYT https://github.com/grafana/explore-logs/pull/628/files#diff-9ef2f732762145f9751be0e3c3cc4ce83b3ad2f7fbff9563aeea16cb67d3033fR62-R64